src/storage/txn/scheduler.rs

```diff
@@ -21,27 +21,31 @@
 //! to the scheduler.
 
 use std::fmt::{self, Debug, Display, Formatter};
+use std::sync::atomic::{AtomicU64, AtomicUsize, Ordering};
+use std::sync::{Arc, Mutex};
 use std::u64;
 
+use futures::future;
 use kvproto::kvrpcpb::CommandPri;
 use prometheus::HistogramTimer;
+use tikv_util::collections::HashMap;
+use tikv_util::worker::ScheduleError;
 
 use crate::storage::kv::Result as EngineResult;
-use crate::storage::Key;
+use crate::storage::txn::latch::{Latches, Lock};
+use crate::storage::txn::process::{
+    execute_callback, notify_scheduler, Executor, MsgScheduler, ProcessResult, Task,
+};
+use crate::storage::txn::sched_pool::SchedPool;
+use crate::storage::txn::Error;
+use crate::storage::{metrics::*, Key};
 use crate::storage::{Command, Engine, Error as StorageError, StorageCb};
-use tikv_util::collections::HashMap;
-use tikv_util::worker::{self, Runnable};
 
-use super::super::metrics::*;
-use super::latch::{Latches, Lock};
-use super::process::{execute_callback, Executor, ProcessResult, Task};
-use super::sched_pool::*;
-use super::Error;
-
-pub const CMD_BATCH_SIZE: usize = 256;
+const TASKS_SLOTS_NUM: usize = 1 << 20;
```

Comment:
1 million locks? I think it is too many, usually several times of the number of total threads is ok, for example 1 << 6 is enough. Please refer to RocksDB's shard block cache.